src/native/libs/System.Security.Cryptography.Native/pal_ssl.c

```diff
@@ -38,6 +38,53 @@ static void EnsureLibSsl10Initialized()
 }
 #endif
 
+#ifdef FEATURE_DISTRO_AGNOSTIC_SSL
+// redirect all SSL_CTX_set_options and SSL_set_options calls via dynamic shims
+// to work around ABI breaking change between 1.1 and 3.0
+
+#undef SSL_CTX_set_options
+#define SSL_CTX_set_options SSL_CTX_set_options_dynamic
+static uint64_t SSL_CTX_set_options_dynamic(SSL_CTX* ctx, uint64_t options)
+{
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wcast-function-type"
+    if (ERR_new) // OpenSSL 3.0 sentinel function
```

Comment:
```suggestion
    if (API_EXISTS(ERR_new)) // OpenSSL 3.0 sentinel function
```

(also up here :smile:)